ROADMAP.md

```diff
@@ -2,14 +2,63 @@
+  - Prioritize the creation of frequently needed components.
+  - Optimize for the people maintaining the components. The more people maintaining the components the better.
+  - Encourage the usage of third-party components if they already exist and are well maintained. We can create [a search experience](https://xpvrpr7r4o.codesandbox.io/) for this.
+- 0.5 - **Better customization.** 💅 We want to make our component customization intuitive, no matter if you are using global CSS or styled-components:
+  - Improve the documentation.
+  - Better support for styled-components.
+  - Work on the core solution.
+  - Provide [more examples like this](https://mui-treasury.com/components/card).
+- 0.3 - **Better documentation.** 📚 It's a wide topic. We don't know exactly what people want. We will focus our effort in the following areas:
+  - Beginner tutorials & Video lessons.
+  - Page Layout Examples. They get people started really quickly, we need more of them!
+  - Page documentation rating 🥇🥈🥉. We will integrate a rating module in all our documentation pages. This way, we can collect high-quality data points and prioritize the pages that need the most to be improved.
+  - Internationalization. Not everybody can read & understand English. The documentation pages are synced with [Crowdin](https://translate.material-ui.com/). They should always be up-to-date.
+- 0.3 - **TypeScript.** 📏 There are two dimensions to this problem:
+  - The documentation. We want to provide a TypeScript variant to most of our demos.
+    It should make it straightforward to use Material-UI with TypeScript. You can help us out in [#14897](https://github.com/mui-org/material-ui/issues/14897).
+  - The definitions. We are **continuously** improving them. We are aware that correctly typing withStyles() is cumbersome. We encourage people to use [makeStyles()](/css-in-js/basics/#hook-api). It should be much better with this Hook API.
+  The codebase is written in JavaScript, we don't plan on migrating it to TypeScript in the near future. Maybe for v5, we will see.
+- 0.2 - **Performance.** 🚀 React abstraction has a cost. The more components you render, the slower your page will be. We can notice stark differences when rendering a large table or list.
+Performance is all about doing less work. We have the following leverages:
+  - Make the core faster. Using our table components is x3 slower than using native table elements.
+    What's the difference? a Material-UI table component creates many **intermediary components**, slowing the rendering.
+    We are working on removing these intermediary components. We are migrating the codebase from class based components to hook based components.
+    [It will yield a theoretical **+30%** performance gain](https://github.com/mui-org/material-ui/issues/10778#issuecomment-472840548), as well as making the DX much better.
```

Comment:
```suggestion
    [This will yield a theoretical **+30%** performance gain](https://github.com/mui-org/material-ui/issues/10778#issuecomment-472840548), as well as making the DX much better.
```